src/nvim/if_cscope.c

```diff
@@ -1685,8 +1685,15 @@ static int cs_read_prompt(size_t i)
   assert(IOSIZE >= cs_emsg_len);
   size_t maxlen = IOSIZE - cs_emsg_len;
 
-  for (;; ) {
-    while ((ch = getc(csinfo[i].fr_fp)) != EOF && ch != CSCOPE_PROMPT[0]) {
+  while (1) {
+    while (1) {
+      do {
+        errno = 0;
+        ch = getc(csinfo[i].fr_fp);
+      } while (errno == EINTR);
```

Comment:
I still think it's best to check `ch == EOF && ferror(csinfo[i].fr_fp) && errno == EINTR` since there's nothing guaranteeing that `errno` remains unchanged on a successful call to `getc`.